fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/observer/SimpleObserver.java

```diff
@@ -68,6 +82,60 @@
     static final Integer EVENT_TYPES = NODE_ADDED + NODE_REMOVED + NODE_MOVED + PROPERTY_ADDED + PROPERTY_CHANGED
             + PROPERTY_REMOVED;
 
+    private static Stream<FedoraResource> getAllChildren(final FedoraResource resource) {
```

Comment:
Possibly? how about a specialization of the existing function like so: `FedoraResource::getChildren(final boolean recursive)`? Then, with no params, you get the direct descendants, and with a `recursive=true` you get all children, recursively.